wallet/pskt/src/bundle.rs

```diff
@@ -1,9 +1,131 @@
-// use crate::role::*;
+use crate::error::Error;
+use crate::pskt::{Inner as PSKTInner, PSKT};
+// use crate::wasm::bundle;
+use hex;
 use serde::{Deserialize, Serialize};
-use std::marker::PhantomData;
+use std::ops::Deref;
 
 #[derive(Debug, Serialize, Deserialize)]
-pub struct Bundle<ROLE> {
-    #[serde(skip_serializing, default)]
-    role: PhantomData<ROLE>,
+pub struct Bundle {
+    pub inner_list: Vec<PSKTInner>,
+}
+
+impl<ROLE> From<PSKT<ROLE>> for Bundle {
+    fn from(pskt: PSKT<ROLE>) -> Self {
+        Bundle { inner_list: vec![pskt.deref().clone()] }
+    }
+}
+
+impl Bundle {
+    pub fn new() -> Self {
+        Self { inner_list: Vec::new() }
+    }
+
+    /// Adds an Inner instance to the bundle
+    pub fn add_inner(&mut self, inner: PSKTInner) {
+        self.inner_list.push(inner);
+    }
+
+    /// Adds a PSKT instance to the bundle
+    pub fn add_pskt<ROLE>(&mut self, pskt: PSKT<ROLE>) {
+        self.inner_list.push(pskt.deref().clone());
+    }
+
+    /// Merges another bundle into the current bundle
+    pub fn merge(&mut self, other: Bundle) {
+        for inner in other.inner_list {
+            self.inner_list.push(inner);
+        }
+    }
+
+    // todo: field support for bincode to directly bincode
+    pub fn to_hex(&self) -> Result<String, Box<dyn std::error::Error>> {
+        // Serialize the bundle to JSON
+        let json_string = serde_json::to_string(self)?;
+
+        // Encode the JSON string to hexadecimal
+        Ok(hex::encode(json_string))
```

Comment:
json can be represented in different ways with padding whitespace, etc
I would prefer cbor or message pack for binary representation

Reply:
Problem is that underlying fields of PSKT do not support it // todo: field support for bincode to directly bincode.

There are underlying data types that cannot be deserialized, serialization mostly work but not the inverse for both:

* rmp serde there's an issue with smallvec and other types, serialize works but deserialize errors: https://github.com/3Hren/msgpack-rust/issues/163

* bincode error: SequenceMustHaveLength

Reply:
I didn't say anything about bincode. Btw it's impossible to implement with serde macroses. Bincode serialization in case of pskt only possible with manual implementation.
I didn't get message pack error. We already using our version of serde-bytes

Reply:
I've added a test that showcases the error when deserializing with rmp-serde.